lexers/bash.lua

```diff
@@ -27,11 +27,20 @@ local sq_str = lexer.range("'", false, false)
 local dq_str = lexer.range('"')
 local ex_str = lexer.range('`')
 local heredoc = '<<' * P(function(input, index)
-  local _, e, _, delimiter = input:find(
-    '^%-?(["\']?)([%a_][%w_]*)%1[\n\r\f;]+', index)
-  if not delimiter then return end
-  _, e = input:find('[\n\r\f]+' .. delimiter, e)
-  return e and e + 1 or #input + 1
+  local s, e, minus, _, delimiter =
+    input:find('(-?)%s*(["\']?)([%w_]+)%2[^\n]*[\n\r\f;]+', index)
```

Comment:
`[%w_]` will also match numbers. Are numbers valid delimiters? I am asking because I don't know all of the intricacies of bash.

What is the reason for `[^\n]*` before the newline/semicolon check? Are there valid characters between a beginning delimiter and the end-of-line? Again, I just don't know all the bash details.